tests/shakedown/shakedown/clients/node.py

```diff
@@ -0,0 +1,132 @@
+        return requests.get(url, *args, **kwargs)
+
+    def post(self, path, *args, **kwargs):
+        url = self.create_url(path)
+        kwargs['auth'] = self.auth
+        return requests.post(url, *args, **kwargs)
+
+
+class DiagnosticBundle():
+
+    def __init__(self, session, bundle_name):
+        logger.info('Created diagnostic bundle %s', bundle_name)
+
+        self.session = session
+        self.bundle_name = bundle_name
+
+    def status(self):
+        """Check the status of the bundle creation.
+
+        :returns None if the bundle is not done yet, the bundle's file name otherwise.
+        """
+        logger.info('Retrieve status for %s', self.bundle_name)
+        resp = self.session.get('list/all')
+        resp.raise_for_status()
+
+        nodes = resp.json()
+
+        # Flatten bundle lists
+        bundle_list = [bundle for bundle_list in nodes.values() if bundle_list is not None for bundle in bundle_list]
+
+        # Search for our bundle.
+        for bundle in bundle_list:
+            file_name = bundle['file_name']
+            if os.path.basename(file_name) == self.bundle_name:
+                return file_name
+
+        return None
+
+    def download(self, bundle_path):
+        file_name = self.status()
+        bundle_name = os.path.basename(file_name)
+        assert file_name is not None, 'The bundle is not ready yet.'
+
+        logger.info('Downloading diagnostic bundle %s', self.bundle_name)
+
+        download_path = os.path.join('serve', bundle_name)
+        with self.session.get(download_path, stream=True) as r:
+            with open(bundle_path, 'wb') as f:
+                for chunk in r.iter_content(1024):
+                    f.write(chunk)
+
+
+class Diagnostics(Collection):
+
+    def __init__(self, session):
+        self.session = session
+
+    def create(self, nodes=None):
+        if nodes is None:
+            nodes = {"nodes": ["all"]}
+        resp = self.session.post('create', json=nodes)
+        resp.raise_for_status()
+        bundle_name = resp.json()['extra']['bundle_name']
+        return DiagnosticBundle(self.session, bundle_name)
+
+    def all(self):
+        resp = self.session.get('list/all')
+        resp.raise_for_status()
+
+        nodes = resp.json()
+
+        # Flatten bundle lists
+        bundle_list = [bundle for bundle_list in nodes.values() if bundle_list is not None for bundle in bundle_list]
+
+        # Yield all bundle names.
+        for bundle in bundle_list:
+            file_name = bundle['file_name']
+            bundle_name = os.path.basename(file_name)
+            yield DiagnosticBundle(self.session, bundle_name)
+
+
+
+class Node(Session):
+    """Resource API session for `/system` endpoint of DC/OS
+
+    Example::
+        client = system.Node()
+
+        diagnostics = client.diagnostics.create()
+
+        bundle = diagnostics.create()
+        while bundle.status() is None:
+            sleep(500)
+
+        bundle.download()
+    """
```

Comment:
@adamtheturtle, @br-lewis and @janisz, this is my first go at defining a different client scheme. I tried to follow the AWS [Boto3 collections pattern](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/collections.html). It's a little strange since the API endpoint is `system` but the CLI calls it `node`. I started with `System` but settle for `Node`. Anyhow, this seems much more straight forward to me. Please also checkout the pattern for apps and pods in #6787.